sdk/keyvault/Azure.Security.KeyVault.Keys/samples/Sample2_BackupAndRestore.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using Azure.Identity;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Threading;
+
+namespace Azure.Security.KeyVault.Keys.Samples
+{
+    /// <summary>
+    /// Sample demonstrates how to backup and restore keys in the key vault
+    /// using the synchronous methods of the KeyClient.
+    /// </summary>
+    [Category("Live")]
+    public partial class BackupAndRestore
+    {
+        [Test]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/6514")]
+        public void BackupAndRestoreSync()
+        {
+            // Environment variable with the Key Vault endpoint.
+            string keyVaultUrl = Environment.GetEnvironmentVariable("AZURE_KEYVAULT_URL");
+            string backupPath = Path.GetTempFileName();
+
+            // Instantiate a key client that will be used to call the service. Notice that the client is using default Azure
+            // credentials. To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
+            // 'AZURE_CLIENT_KEY' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+            var client = new KeyClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
+
+            // Let's create a Rsa key valid for 1 year. If the key
+            // already exists in the key vault, then a new version of the key is created.
+            string rsaKeyName = $"CloudRsaKey-{Guid.NewGuid()}";
+            var rsaKey = new RsaKeyCreateOptions(rsaKeyName, hsm: false, keySize: 2048)
+            {
+                Expires = DateTimeOffset.Now.AddYears(1)
+            };
+
+            Key storedKey = client.CreateRsaKey(rsaKey);
+
+            // Backups are good to have if in case keys get accidentally deleted by you.
+            // For long term storage, it is ideal to write the backup to a file.
+            File.WriteAllBytes(backupPath, client.BackupKey(rsaKeyName));
+
+            // The storage account key is no longer in use, so you delete it.
+            client.DeleteKey(rsaKeyName);
+
+            // To ensure the key is deleted on server side.
+            Assert.IsTrue(WaitForDeletedKey(client, rsaKeyName));
+
+            // If the keyvault is soft-delete enabled, then for permanent deletion, deleted key needs to be purged.
+            client.PurgeDeletedKey(rsaKeyName);
+
+            // After sometime, the key is required again. We can use the backup value to restore it in the key vault.
+            KeyBase restoredKey = client.RestoreKey(File.ReadAllBytes(backupPath));
+
+            AssertKeysEqual((KeyBase)storedKey, restoredKey);
```

Comment:
You may want to consider having a `try/finally` that removes the temp file at the end of this;  you're making changes to a caller's machine and leaving orphaned files, currently.   I may also suggest not emitting a file, since you can't be 100% sure that the caller has permission to write to the file system;  I think a `MemoryStream` would illustrate the concept without altering the caller's environment.

Reply:
I've never used MemoryStream. I will give it a try :) Looks great!

Reply:
Changes done. @jsquire could you take a look at this part again to make sure I am using `MemoryStream` as expected?
I tested it locally (by deleting the `Ignore` and setting a time delay of 1min and it works .. but still

Reply:
Looks solid to me.   You may want to add a comment or two just to call out that you're storing in a temporary memory area, but the normal case scenario would be to write the backup out to disk or to a database or other durable storage.

Reply:
Thanks Jesse!